spec/lib/services/account-api-service-spec.js

```diff
@@ -0,0 +1,154 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+require('../../helper');
+
+describe("Account API Service", function() {
+    var sandbox;
+    var self = this;
+    var waterline;
+    var accountService;
+    var Promise;
+
+    var user1 = {
+        id: 0,
+        username: 'admin',
+        password: 'admin123',
+        role: 'Administrator'
+    };
+    
+    var user2 = {
+        username: 'readonly',
+        password: 'read123',
+        role: 'ReadOnly'
+    };
+
+    before(function() {
+        helper.setupInjector([
+            helper.require("/lib/services/account-api-service"),
+            dihelper.simpleWrapper(function() { arguments[1](); }, 'rimraf')
+        ]);
+
+        waterline = helper.injector.get('Services.Waterline');
+        waterline.localusers = {
+            find: function() {},
+            findOne: function() {},
+            create: function() {},
+            update: function() {},
+            destroy: function() {}
+        };
+        accountService = helper.injector.get('Http.Services.Api.Account');
+        Promise = helper.injector.get('Promise');
+
+        self.sandbox = sinon.sandbox.create();
+        self.sandbox.stub(waterline.localusers, 'find');
+        self.sandbox.stub(waterline.localusers, 'findOne');
+        self.sandbox.stub(waterline.localusers, 'create');
+        self.sandbox.stub(waterline.localusers, 'update');
+        self.sandbox.stub(waterline.localusers, 'destroy');
+
+        // Setup ACL rules that are missed during startServer
+        return Promise.all([
+            accountService.aclMethod('addUserRoles', 'admin', 'Administrator'),
+            accountService.aclMethod('addUserRoles', 'readonly', 'ReadOnly'),
+            accountService.aclMethod('addRoleParents', 'Administrator', ['ConfigureUsers']),
+            accountService.aclMethod('addRoleParents', 'ReadOnly', ['ConfigureSelf'])
+        ]);
+    });
+
+    after(function() {
+        self.sandbox.restore();
+    });
+
+    afterEach(function() {
+        self.sandbox.reset();
+    });
+
+    it('should list users', function() {
+        waterline.localusers.find.resolves([user1, user2]);
+        return accountService.listUsers()
+        .then(function(users) {
+            expect(waterline.localusers.find).to.have.been.called;
+            expect(users.length).to.equal(2);
+            return Promise.map(users, function(user) {
+                expect(user.username).to.be.a('string');
+                expect(user.password).to.be.an('undefined');
+                expect(user.role).to.be.a('string');
+                expect(user.privileges).to.be.instanceof(Array);
```

Comment:
It will be better to expect the user strictly equals to the info of `user1` and `user2` rather than only judging the returned types,